tools/pythonpkg/src/include/duckdb_python/vector_conversion.hpp

```diff
@@ -47,6 +47,8 @@ struct PandasColumnBindData {
 	unique_ptr<NumPyArrayWrapper> mask;
 	// Only for categorical types
 	string internal_categorical_type;
+	// When object types are casted we must hold their data somewhere
+	vector<py::str> object_str_val;
```

Comment:
Should we grab the gil in the destructor of the PandasColumnBindData?

Reply:
100%, I assumed it was being held by whoever calls its destructor since it already had a py::array in there.

Maybe we should double-check all these python wrappers, they must acquire the gil before constructing/destructing py objects.